src/routes/safe/components/Apps/trackAppUsageCount.ts

```diff
@@ -0,0 +1,64 @@
+import { SafeApp } from 'src/routes/safe/components/Apps/types'
+import local from 'src/utils/storage/local'
+
+export const APPS_DASHBOARD = 'APPS_DASHBOARD'
+
+const TX_COUNT_WEIGHT = 2
+const OPEN_COUNT_WEIGHT = 1
+const MORE_RECENT_MULTIPLIER = 2
+const LESS_RECENT_MULTIPLIER = 1
+
+export type AppTrackData = {
+  [safeAppId: string]: {
+    timestamp: number
+    openCount: number
+    txCount: number
+  }
+}
+
+export const getAppsUsageData = (): AppTrackData => {
+  return local.getItem<AppTrackData>(APPS_DASHBOARD) || {}
+}
+
+export const trackSafeAppOpenCount = (id: SafeApp['id']): void => {
+  const trackData = getAppsUsageData()
+  const currentOpenCount = trackData[id]?.openCount || 0
+  const currentTxCount = trackData[id]?.txCount || 0
+
+  local.setItem(APPS_DASHBOARD, {
+    ...trackData,
+    [id]: {
+      timestamp: Date.now(),
+      openCount: currentOpenCount + 1,
+      txCount: currentTxCount,
+    },
+  })
+}
+
+export const trackSafeAppTxCount = (id: SafeApp['id']): void => {
+  const trackData = getAppsUsageData()
+  const currentTxCount = trackData[id]?.txCount || 0
+
+  local.setItem(APPS_DASHBOARD, {
+    ...trackData,
+    [id]: { ...trackData[id], txCount: currentTxCount + 1 },
```

Comment:
I assume it will always have an `openCount` here as the App has opened?

Reply:
I'll add a comment